lib/errors.js

```diff
@@ -285,6 +285,9 @@ internals.annotate = function (stripColorCodes) {
                         delete ref[replacement];
                     }
                     else {
+                        if (typeof ref === 'string') {
```

Comment:
Just add some more nesting on `a` and cause a deeper error, like `a: "{b:{c:"string"}}". Your code basically only works if the error happens on the object being parsed, not inside.